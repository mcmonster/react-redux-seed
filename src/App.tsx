/**
 * @blame R. Matt McCann <me@mattmccann.xyz>
 * @brief Entry point for the app
 */

import { ConnectedRouter } from 'connected-react-router';
import * as React from 'react';
import { Provider } from 'react-redux';
import { Route, Switch } from 'react-router';

import './App.css';
import logo from './logo.svg';
import { configureStore, history } from './redux/configure-store';


// Build the store we will use with the app
const store = configureStore();


const match = () => <div>Match</div>;
const miss = () => <div>Miss</div>;

class App extends React.Component {
  public render() {
    return (
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <div className="App">
            <header className="App-header">
              <img src={logo} className="App-logo" alt="logo" />
              <h1 className="App-title">Welcome to React</h1>
            </header>
            <p className="App-intro">
              To get started, edit <code>src/App.tsx</code> and save to reload.
            </p>
            <Switch>
              <Route exact path="/" component={match} />
              <Route component={miss} />
            </Switch>
          </div>
        </ConnectedRouter>
      </Provider>
    );
  }
}

export default App;
