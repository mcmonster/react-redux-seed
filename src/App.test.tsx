import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';

import App from './App';


describe('App', () => {
  let component: ShallowWrapper<App>;

  beforeEach(() => {
    component = shallow(<App />);
  });

  it('renders without crashing', () => {
    expect(component).toBeTruthy();
  });

  it('renders as expected', () => {
    expect(component).toMatchSnapshot();
  });
});
