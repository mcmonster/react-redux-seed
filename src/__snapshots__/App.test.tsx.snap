// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ConnectedRouter)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <div
      className="App"
    >
      <header
        className="App-header"
      >
        <img
          alt="logo"
          className="App-logo"
        />
        <h1
          className="App-title"
        >
          Welcome to React
        </h1>
      </header>
      <p
        className="App-intro"
      >
        To get started, edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </div>
  </Connect(ConnectedRouter)>
</Provider>
`;
